sotodlib/preprocess/processes.py

```diff
@@ -57,32 +60,50 @@ def select(self, meta):
         meta.restrict("dets", meta.dets.vals[keep])
         return meta
 
+    @staticmethod
+    def gen_metric(cls, meta, proc_aman):
+        # For now just compute the fraction of samples that were flagged
+        # (I think in practice this will be 1 or 0...)
+        # TODO: come up with a more useful metric
+        frac_flagged = np.array([
+            np.dot(r.ranges(), [-1, 1]).sum() / meta.obs_info.n_samples
+            for r in proc_aman.det_bias_flags
+        ])
+        obs_time = [meta.obs_info.timestamp] * frac_flagged.size
```

Comment:
just noting that the only reason to pass the `meta` object is to extract this timestamp, but in the new scheme this could probably be done in the `QAMetric` method.

Reply:
I think this is a totally legitimate way to do this but just to point out as you're writing QA metrics from flags there is a method in the `RangesMatrix` class already called `get_stats()` which you could use. I think the only place it appears in sotodlib at the moment is in `flag_utils` here: https://github.com/simonsobs/sotodlib/blob/60a63dcef338dad7be4ad6d02d5fb348f4cc636c/sotodlib/flag_utils.py#L51